marine_benchmark.c

```diff
@@ -137,14 +138,15 @@ int main(int argc, char *argv[]) {
             "frame.encap_type"
     };
 
+
     benchmark_case cases[] = {
-            {"Benchmark with BPF",                                            bpf, NULL,    NULL, 0},
-            {"Benchmark with Display filter",         NULL,                        dfilter, NULL, 0},
-            {"Benchmark with BPF and Display filter",                         bpf, dfilter, NULL, 0},
-            {"Benchmark with three extracted fields", NULL,                        NULL,    three_fields, ARRAY_SIZE(three_fields)},
-            {"Benchmark with eight extracted fields", NULL,                        NULL,    eight_fields, ARRAY_SIZE(eight_fields)},
-            {"Benchmark with BPF, Display filter and three extracted fields", bpf, dfilter, three_fields, ARRAY_SIZE(three_fields)},
-            {"Benchmark with BPF, Display filter and eight extracted fields", bpf, dfilter, eight_fields, ARRAY_SIZE(eight_fields)},
+            {"Benchmark with BPF",                                            bpf,  NULL,    NULL,         NULL,           0},
+            {"Benchmark with Display filter",                                 NULL, dfilter, NULL,         NULL,           0},
+            {"Benchmark with BPF and Display filter",                         bpf,  dfilter, NULL,         NULL,           0},
+            {"Benchmark with three extracted fields",                         NULL, NULL,    three_fields, NULL, ARRAY_SIZE(three_fields)},
+            {"Benchmark with eight extracted fields",                         NULL, NULL,    eight_fields, NULL, ARRAY_SIZE(eight_fields)},
+            {"Benchmark with BPF, Display filter and three extracted fields", bpf,  dfilter, three_fields, NULL, ARRAY_SIZE(three_fields)},
+            {"Benchmark with BPF, Display filter and eight extracted fields", bpf,  dfilter, eight_fields, NULL, ARRAY_SIZE(eight_fields)},
```

Comment:
Very nice!
Optionally, I think you could add a benchmark case with macros (similar to the one you added in Python).